Public/Src/Utilities/Configuration/ExecutionEnvironment.cs

```diff
@@ -19,99 +22,134 @@ public enum ExecutionEnvironment
 
         /// <summary>
         /// Office enlistment-build on build lab.
         /// </summary>
-        OfficeEnlistmentBuildLab,
+        OfficeEnlistmentBuildLab = Office | EnlistBuild | Lab,
 
         /// <summary>
         /// Office meta-build on developer machine.
         /// </summary>
-        OfficeMetaBuildDev,
+        OfficeMetaBuildDev = Office | MetaBuild | Dev,
 
         /// <summary>
         /// Office meta-build on build lab.
         /// </summary>
-        OfficeMetaBuildLab,
+        OfficeMetaBuildLab = Office | MetaBuild | Lab,
 
         /// <summary>
         /// Office product-build on developer machine.
         /// </summary>
-        OfficeProductBuildDev,
+        OfficeProductBuildDev = Office | ProductBuild | Dev,
 
         /// <summary>
         /// Office product-build (or regular-build) on build lab.
         /// </summary>
-        OfficeProductBuildLab,
+        OfficeProductBuildLab = Office | ProductBuild | Lab,
 
         //////////////// Visual C++ environment ///////////////////////
 
         /// <summary>
         /// Visual cpp team tests as run in the lab
         /// </summary>
-        VisualCppTestsLab,
+        VisualCppTestsLab = VisualCpp | Tests | Lab,
 
         /// <summary>
         /// Visual cpp team tests as run by devs
         /// </summary>
-        VisualCppTestsDev,
+        VisualCppTestsDev = VisualCpp | Tests | Dev,
 
         ///////////////// Office macOS //////////////////////////
 
         /// <summary>
         /// Office APEX build in the build lab
         /// </summary>
-        OfficeAPEXLab,
+        OfficeAPEXLab = OfficeApex | NoStage | Lab,
 
         /// <summary>
         /// Office APEX build on a dev machine
         /// </summary>
-        OfficeAPEXDev,
+        OfficeAPEXDev = OfficeApex | NoStage | Dev,
 
         ///////////////////////////////////////////////////////////////
     }
+
+    /// <summary>
+    /// Defines components of <see cref="ExecutionEnvironment"/> enum values.
+    /// Values should be mutually exclusive for their bit range and not set any bits in other bit ranges.
+    /// 
+    /// 1st and 2nd hex chars (8 bits) is product group
+    /// 3rd hex char (4 bits) is build stage
+    /// 4th hex char (4 bits) is build location
+    /// </summary>
+    internal enum ExecutionEnvironmentParts
+    {
+        // Group 0xXX__
+        SelfHost   = 0xBD00,
+        Office     = 0x0F00,
+        OfficeApex = 0x0A00,
+        OsgTools   = 0x0200,
+        Osg        = 0x0500,
+        VisualCpp  = 0x7C00,
+
+        // Stage 0x__X0
+        NoStage      = 0x0000,
+        EnlistBuild  = 0x0010,
+        MetaBuild    = 0x0020,
+        ProductBuild = 0x0030,
+        Tests        = 0x0040,
+        WrapItUp     = 0x0050,
+        Tools        = 0x0060,
+        NightlyPerf  = 0x0070,
+
+        // Location 0x___X
+        NoLocation = 0x0000,
+        Lab        = 0x0007,
+        PrimeLab   = 0x0009,

```

Comment:
>     PrimeLab   = 0x0009, [](start = 3, length = 25)

No need to differentiate this. It doesn't appear that anyone's passing the OsgPrimeLab flag anymore. Don't delete the enum for sake of not breaking back compat, but i think we can keep the locations simple